instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/utils/ApproximateCoordinates.kt

```diff
@@ -0,0 +1,37 @@
+package com.mapbox.navigation.instrumentation_tests.utils
+
+import com.mapbox.geojson.Point
+import kotlin.math.abs
+
+class ApproximateCoordinates(
+    val latitude: Double,
+    val longitude: Double,
+    private val tolerance: Double,
+) {
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as ApproximateCoordinates
+
+        if (abs(latitude - other.latitude) > tolerance) return false
+        if (abs(longitude - other.longitude) > tolerance) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        var result = latitude.hashCode()
+        result = 31 * result + longitude.hashCode()
+        return result
+    }
```

Comment:
as I understand two approximate coordinates could be equals, but their hash codes will be different. In this case data structures like hash tables won't be able to find equal objects. Not sure if it's important problem, but this could be unexpected

Reply:
Hmmm, I think you're right. It's not a problem now because it's used from `assertEquals`.
But I don't see the point of using `ApproximateCoordinates` as a key in hash table. Because for 2 different coordinates you'll get the same key... However, I think it would make sense to just return 0 here to avoid this unexpectedness.